rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/customheaderauth/Sw360CustomHeaderAuthenticationFilter.java

```diff
@@ -0,0 +1,202 @@
+            throws IOException, ServletException {
+
+        if (active) {
+            CustomHeaderAuthRequestDetails requestDetails = extractRequestDetails((HttpServletRequest) request);
+            if (canAuthenticate(requestDetails)) {
+
+                ServletRequest wrappedRequest = doAuthenticate((HttpServletRequest) request, requestDetails);
+                if (wrappedRequest != null) {
+                    chain.doFilter(wrappedRequest, response);
+                    return;
+                }
+            }
+        }
+
+        chain.doFilter(request, response);
+    }
+
+    private CustomHeaderAuthRequestDetails extractRequestDetails(HttpServletRequest request) {
+        CustomHeaderAuthRequestDetails result = new CustomHeaderAuthRequestDetails();
+
+        result.currentUser = SecurityContextHolder.getContext().getAuthentication();
+        result.customHeaderEmail = StringUtils.defaultIfEmpty(request.getHeader(customHeaderHeadernameEmail), "");
+        result.customHeaderExtId = StringUtils.defaultIfEmpty(request.getHeader(customHeaderHeadernameExtid), "");
+        result.customParamClientId = request.getParameter(OAuth2Utils.CLIENT_ID);
+        result.customParamClientSecret = request.getParameter(PARAMETER_NAME_CLIENT_SECRET);
+        result.grantType = request.getParameter(OAuth2Utils.GRANT_TYPE);
+
+        return result;
+    }
+
+    private boolean canAuthenticate(CustomHeaderAuthRequestDetails requestDetails) {
+        // we want to be active only for grant type 'password' requests
+        if (!requestDetails.grantType.equals("password")) {
+            log.debug("Cannot create authentication object because grant type is not password!");
+            return false;
+        }
+
+        // if there is already authentication information available, do nothing
+        if (requestDetails.currentUser != null) {
+            log.debug("Cannot create authentication object because there is already one!");
+            return false;
+        }
+
+        // without any auth header, this authentication makes no sense
+        if (StringUtils.isEmpty(requestDetails.customHeaderEmail)
+                && StringUtils.isEmpty(requestDetails.customHeaderExtId)) {
+            log.debug("Cannot create authentication object because user identifying headers from proxy are missing!");
+            return false;
+        }
+
+        // without client info, this authentication makes no sense
+        if (StringUtils.isEmpty(requestDetails.customParamClientId)
+                || StringUtils.isEmpty(requestDetails.customParamClientSecret)) {
+            log.debug("Cannot create authentication object because client identifying request parameters are missing!");
+            return false;
+        }
+
+        return true;
+    }
+
+    private ServletRequest doAuthenticate(HttpServletRequest request, CustomHeaderAuthRequestDetails requestDetails) {
```

Comment:
Agreed, it would be nicer. But if I change it now in this location, I would have to change it in several others. Maybe we can leave it like that since it is only a private method and no API?